app/templates/terraform-main.tf_template

```diff
@@ -0,0 +1,142 @@
+# Configure Azure provider
+provider "azurerm" {
+  subscription_id = "${var.azure_subscription_id}"
+  client_id       = "${var.azure_client_id}"
+  client_secret   = "${var.azure_client_secret}"
+  tenant_id       = "${var.azure_tenant_id}"
+}
+
+# create a resource group if it doesn't exist
+resource "azurerm_resource_group" "rg" {
+    name = "{{host.name}}rg"
```

Comment:
The hostname needs to be unique within a resource group. Since we are creating one resource group for each new host, this should be ok. However I'm using the hostname to make up the resource group name and I shouldn't be doing that.

The resource group needs to be unique within the subscription and since everyone is sharing the same subscription for now, we'll need to check for uniqueness once we have a DB in place, or with the Azure Python SDK directly checking on our subscription resources.

The DNS prefix goes in a different part of the host object, and also needs to be checked for uniqueness within all CloudLabs as you say. 

I'll keep track of this on an issue. Will come back to it later on or once we have a DB in place.